lib/livebook/session.ex

```diff
@@ -46,7 +46,17 @@ defmodule Livebook.Session do
   # The struct holds the basic session information that we track
   # and pass around. The notebook and evaluation state is kept
   # within the process state.
-  defstruct [:id, :pid, :origin, :notebook_name, :file, :images_dir, :created_at, :memory_usage]
+  defstruct [
+    :id,
+    :pid,
+    :origin,
+    :notebook_name,
+    :file,
+    :images_dir,
+    :created_at,
+    :memory_usage,
+    :secrets
+  ]
```

Comment:
The session struct should only hold metadata that we need in places other than the session page. I think we want to store the secrets in Session.Data, with everything else.

Then we also need a Session.Data operation for adding a secret, and we can update the runtime in `Session.after_operation`.

Reply:
Gotcha! Thanks!